packages/hardhat-test-helpers/src/chai/matchers/bigNumber.ts

```diff
@@ -0,0 +1,217 @@
+// TODO: consider adding support for the following Chai.Assert properties.
+// Might any of these come "for free" due to supporting other methods? At the
+// very least get them tested.
+//   isAbove
+//   isAtLeast
+//   isBelow
+//   isAtMost
+//   isAt
+//   isNumber
+//   isNotNumber
+//   closeTo
+//   approximately
+
+import { BigNumber as BigNumberEthers } from "ethers";
+import { BigNumber as BigNumberJs } from "bignumber.js";
+import BN from "bn.js";
+
+export type SupportedNumber =
+  | number
+  | bigint
+  | BN
+  | BigNumberEthers
+  | BigNumberJs;
+
+export function supportBigNumber(
+  Assertion: Chai.AssertionStatic,
+  utils: Chai.ChaiUtils
+) {
+  const equalsFunction = override("eq", "equal", "not equal", utils);
+  Assertion.overwriteMethod("equals", equalsFunction);
+  Assertion.overwriteMethod("equal", equalsFunction);
+  Assertion.overwriteMethod("eq", equalsFunction);
+
+  const gtFunction = override("gt", "be above", "be at most", utils);
+  Assertion.overwriteMethod("above", gtFunction);
+  Assertion.overwriteMethod("gt", gtFunction);
+  Assertion.overwriteMethod("greaterThan", gtFunction);
+
+  const ltFunction = override("lt", "be below", "be at least", utils);
+  Assertion.overwriteMethod("below", ltFunction);
+  Assertion.overwriteMethod("lt", ltFunction);
+  Assertion.overwriteMethod("lessThan", ltFunction);
+
+  const gteFunction = override("gte", "be at least", "be below", utils);
+  Assertion.overwriteMethod("least", gteFunction);
+  Assertion.overwriteMethod("gte", gteFunction);
+  Assertion.overwriteMethod("greaterThanOrEqual", gteFunction);
+
+  const lteFunction = override("lte", "be at most", "be above", utils);
+  Assertion.overwriteMethod("most", lteFunction);
+  Assertion.overwriteMethod("lte", lteFunction);
+  Assertion.overwriteMethod("lessThanOrEqual", lteFunction);
+
+  Assertion.overwriteMethod("within", overrideWithin(utils));
+
+  Assertion.overwriteMethod("closeTo", overrideCloseTo(utils));
+}
+
+type Methods = "eq" | "gt" | "lt" | "gte" | "lte";
+
+function override(
+  method: Methods,
+  name: string,
+  negativeName: string,
+  utils: Chai.ChaiUtils
+) {
+  return (_super: (...args: any[]) => any) =>
+    overwriteBigNumberFunction(method, name, negativeName, _super, utils);
+}
+
+function normalize(source: SupportedNumber | string): bigint {
+  if (source instanceof BigNumberEthers) {
```

Comment:
This doesn't work if the user has a different version of ethers (or of the other libraries). I think this is another good reason to use another approach (again, maybe based in optional dependencies, or in `any`ing everything).

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/2428/commits/b317f4d52a2c7be29f913d3292e1bcf0e7c862c3